src/fides/api/service/connectors/sql_connector.py

```diff
@@ -184,15 +185,26 @@ def mask_data(
         query_config = self.query_config(node)
         update_ct = 0
         client = self.client()
+
+        # Check if safe_mode is enabled
+        db: Session = Session.object_session(self.configuration)
+        config_proxy = ConfigProxy(db)
+        safe_mode_enabled = getattr(config_proxy.execution, "safe_mode", False)
```

Comment:
I rarely suggest this (I'm not much of a lover of inheritance), but this does seem to be a scenario where this could be defined on the BaseConnector class and reused for SqlConnector, BigQueryConnector, etc.

Reply:
It looks like most of the other classes are using Depends(get_config_proxy) which would tidy this up and make it more consistent with how the rest of the code gets configs. I'll clean this up and the other instance

Reply:
Ok I did end up using inheritance since this was the base class and the other standard patterns didn't apply (CONFIG only  works for static configs and Depends only works on the FastAPI endpoints themselves)